### PR TITLE
feat(client): support retries and timeout for pypi settings

### PR DESCRIPTION
**Pull request by tianweidut**
## Description
support timeout and retries for pypi

## Modules
- [x] Controller
- [x] Client

## Checklist
- [x] run code format and lint check
- [x] add unit test
- [ ] add necessary doc

### CODE REVIEW

File: client/scripts/sw-docker-entrypoint [hunk 1]
```diff
@@ -39,6 +39,7 @@ pre_config() {
 [global]
 index-url = ${SW_PYPI_INDEX_URL}
 extra-index-url = ${SW_PYPI_EXTRA_INDEX_URL}
+timeout = ${SW_PYPI_TIMEOUT:-90}
```
Comment (goldenxinxing) [inline client/scripts/sw-docker-entrypoint:42]: Is there has a ‘RETRIES’ config? and add it for the following command of `python3 -m pip install "starwhale==${DETECTED_SW_VER}"`

Comment (tianweidut) [reply client/scripts/sw-docker-entrypoint:42]: pip.conf does not include retries config.

Comment (goldenxinxing) [reply client/scripts/sw-docker-entrypoint:42]: ok

**Review by goldenxinxing — APPROVED** [review]
LGTM

**Pull request merged**